src/Akeneo/Connectivity/Connection/back/Infrastructure/Apps/Persistence/Query/GetConnectedAppIdAndUserGroupQuery.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Infrastructure\Apps\Persistence\Query;
+
+use Akeneo\Connectivity\Connection\Domain\Apps\Persistence\Query\GetConnectedAppIdAndUserGroupQueryInterface;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GetConnectedAppIdAndUserGroupQuery implements GetConnectedAppIdAndUserGroupQueryInterface
+{
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function execute(string $appPublicId): ?array
+    {
+        $selectQuery = <<<SQL
+SELECT acca.id, oag.name
+FROM pim_api_client pac
+JOIN akeneo_connectivity_connection acc on pac.id = acc.client_id
+JOIN akeneo_connectivity_connected_app acca on acc.code = acca.connection_code
+JOIN oro_user_access_group ouag on acc.user_id = ouag.user_id
+JOIN oro_access_group oag on ouag.group_id = oag.id
+WHERE pac.marketplace_public_app_id = :id AND oag.name != 'All'
+SQL;
+        $resultRow = $this->connection->executeQuery($selectQuery, ['id' => $appPublicId])->fetch() ?: null;
```

Comment:
```suggestion
        $stmt = $this->connection->executeQuery($selectQuery, ['id' => $appPublicId]);

        if (false === $row = $stmt->fetch()) {
          return null;
        }

        return $row;
```
What do you think ?
Also, instead of mapping the keys, I usually rename the columns directly in my query.
Like `SELECT id AS appId, ...`
It's not required, only a suggestion.